dial.go

```diff
@@ -89,99 +89,77 @@ func (d *Dialer) Dial(ctx context.Context, raddr ma.Multiaddr, remote peer.ID) (
 		return nil, ipnet.ErrNotInPrivateNetwork
 	}
 
-	var connOut iconn.Conn
```

Comment:
If we're going to do this this way, `err` should be a named return value. Otherwise, directly returning an error won't set `err` correctly.

Reply:
I was using a local variable (which yeah, is not super comfortable) not to make the godoc stutter, but it's a subjective tradeoff. I'll switch.